src/expressions/operators/boolean/AndOperator.ts

```diff
@@ -11,8 +11,45 @@ import Expression from '../../Expression';
 import Specificity from '../../Specificity';
 import { DONE_TOKEN, ready } from '../../util/iterators';
 
+const subBucketsByBucket: Record<string, string[]> = {
```

Comment:
Maybe comment what this expresses. I don't think the name `subBucketsByBucket` indicates very clearly that it gives you buckets which are more specific